mongo/integration/sessions_test.go

```diff
@@ -341,6 +342,21 @@ func TestSessions(t *testing.T) {
 			assertCollectionCount(mt, int64(numDocs))
 		})
 	})
+
+	unackWcOpts := options.Collection().SetWriteConcern(writeconcern.New(writeconcern.W(0)))
+	mt.RunOpts("unacknowledged write", mtest.NewOptions().CollectionOptions(unackWcOpts), func(mt *mtest.T) {
+		// unacknowledged write during a session should result in an error
+		sess, err := mt.Client.StartSession()
+		assert.Nil(mt, err, "StartSession error: %v", err)
+		defer sess.EndSession(context.Background())
+
+		err = mongo.WithSession(context.Background(), sess, func(sc mongo.SessionContext) error {
+			_, err := mt.Coll.InsertOne(sc, bson.D{{"x", 1}})
+			return err
+		})
+
+		assert.Equal(mt, err, mongo.ErrUnacknowledgedWrite, "expected error on unacknowledge session write")
```

Comment:
Optional: Note that this file imports the `"go.mongodb.org/mongo-driver/internal/testutil/assert"` package, not the `"github.com/stretchr/testify/assert"` package. As a result, the printed failure message doesn't include the actual value comparison failure. Consider adding the actual error value to the error message.

E.g.
```go
assert.Equal(
	mt,
	err,
	mongo.ErrUnacknowledgedWrite,
	"expected ErrUnacknowledgedWrite on unacknowledge session write, got %v",
	err)
```